manifests/init.pp

```diff
@@ -63,4 +63,23 @@
   $syslog_protocol    = 'udp',
   $syslog_port        = 514,
   $syslog_facility    = 'local7',
-) { }
+) {
+  
+  # The hackery is strong.  Not sure where else to hack this in.
+  if $::osfamily == "RedHat" {
+    file{'/etc/facter':
+      ensure  => 'directory',
+      owner   => 'root',
+      group   => 'root',
+      mode    => '0444'
+    }
+
+    file{'/etc/facter/facts.d':
+      ensure  => 'directory',
+      owner   => 'root',
+      group   => 'root',
+      mode    => '0444',
+      require => File['/etc/facter']
+    }
+  }
+}
```

Comment:
In the `st2workroom`, I was faced with a similar challenge. I ended up importing a module https://github.com/StackStorm/st2workroom/blob/master/Puppetfile#L32 (ghoneycutt/facter) and just `include ::facter`'d